src/sentry/toolbar/views/base.py

```diff
@@ -0,0 +1,35 @@
+from typing import Any
+
+from django.http import HttpRequest
+
+from sentry.models.organization import Organization
+from sentry.models.project import Project
+from sentry.toolbar.utils.csp import csp_add_directive
+from sentry.web.frontend.base import OrganizationView
+
+
+class ToolbarView(OrganizationView):
+    def respond(self, template: str, context: dict[str, Any] | None = None, status: int = 200):
+        response = super().respond(template, context=context, status=status)
+        # Allow running the inline scripts in the response templates
+        response["Content-Security-Policy"] = csp_add_directive(
+            response.get("Content-Security-Policy", ""),
+            "script-src",
+            ["'unsafe-inline'", "sentry.io", "*.sentry.io"],
```

Comment:
hard-coded domains, this will need to be tweaked to support on-prem installs.

Reply:
Should this use the `sentry:toolbar_allowed_origins` option?

Reply:
I think it'll endup being the domain name(s) of whereever sentry is hosted. That option should be the list of 3rd party domains where their app is running